.github/workflows/testing.yml

```diff
@@ -5,6 +5,11 @@ on: push
 jobs:
   run-test-code:
     runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/bquan0/project-name:latest
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
     strategy:
       matrix:
         python-version: ["3.7", "3.8", "3.9", "3.10"]
```

Comment:
Your image may not have the right versions of `numpy`, `pytest`, `coverage`, etc, for this matrix.  You could make 4 different images by also including the matrix in your `buildImage` workflow, and embed the python version in the image name/tags.